doc: update function naming convention (#3176)
projectacrn/acrn-hypervisor: doc/developer-guides/c_coding_guidelines.rst

   enough with prior rules, it shall be named with the module name as prefix, such
   as 'vie_read_mmio'.
5) If the function is a wrapper of inline Assembly codes, it shall be named with
   one of the following formats:

   a) asm_<Assembly instruction mnemonic>, such as 'asm_pause'.

>    b) If the Assembly instruction mnemonic could not clearly indicate the

Reviewer: ```suggestion
   b) If the Assembly instruction mnemonic does not clearly indicate the
```

      purpose of the function or the function includes multiple Assembly

>       instruction statements, the function shall be named with 'asm' as

Reviewer: ```suggestion
      instruction statements, the function shall be named with 'asm_' as
```

Author: Hi @dbkinder,
Do you think we should update the following two rules as well?

From:
C-NC-03: Data structures exposed to external components shall be named with prefix ‘acrn’
C-NC-04: Data structures only used by hypervisor shall be named with prefix ‘hv’
To:
C-NC-03: Data structures exposed to external components shall be named with prefix ‘acrn_’
C-NC-04: Data structures only used by hypervisor shall be named with prefix ‘hv_’

Reviewer: Yes, that would be a good change (to add the underscore), but be sure to escape the underscore so it doesn't look like a link:  'hv\_'

>       prefix and apply the prior rules.

Reviewer: ```suggestion
      prefix and apply the other non-assembly function naming rules.
```
